Create rule S6394: Unnecessary escaping should be prevented by using other delimiters (#643)
SonarSource/rspec: rules/S6394/php/rule.adoc

> In PHP, PCRE functions, such as `preg_match`, are often used to perform regular expression operations. It is necessary to enclose the actual pattern with delimiters. The delimiters are usually `/`. This leads to the fact that each character must be escaped identically to the delimiter in the pattern.

Reviewer: ```suggestion
In PHP, PCRE functions, such as `preg_match`, are often used to perform regular expression operations. It is necessary to enclose the actual pattern with delimiters. The delimiters are usually `/`. Every occurrence of the delimiter character in the pattern has to be escaped with a backslash.
```

However, delimiters may also be any non-alphanumeric, non-backslash or non-whitespace character.

> In addition to two identical delimiters, it is also allowed to use bracket style delimiters with the opening and closing brackets. This set of different delimiters allows to avoid unnecessary escaping of used characters.

Reviewer: ```suggestion
In addition to two identical delimiters, it is also allowed to use bracket style delimiters with the opening and closing brackets. Using different delimiters out of these possibilities allows avoiding unnecessary escaping, and increases the readability.
```
